capellambse_context_diagrams/collectors/tree_view.py

```diff
@@ -208,6 +207,17 @@ def get_all_classes(
                 classes.update(
                     dict(get_all_classes(prop.type, partition, classes))
                 )
+
+    for cls in root.sub:
+        if cls.is_primitive:
+            continue
+
+        edge_id = f"{root.uuid} {cls.uuid}"
+        if edge_id not in classes:
+            classes[edge_id] = _make_class_info(
+                root, prop, partition, generalizes=cls
+            )
+            classes.update(dict(get_all_classes(cls, partition, classes)))
```

Comment:
You don't need to accumulate this into a dict first, you can directly feed an Iterator of 2-tuples into `dict.update()`.